DatadogInternal/Sources/DatadogCoreProtocol.swift

```diff
@@ -223,6 +223,10 @@ public protocol FeatureScope {
     ///
     /// - Parameter block: The block to execute; it is called on the context queue.
     func context(_ block: @escaping (DatadogContext) -> Void)
+
+    /// Retrieve data store.
+    /// - Returns: Data store configured for storing data for this feature.
+    func dataStore() -> DataStore
```

Comment:
The presence of `DataStore` in `FeatureScope` interface [limits its usage](https://github.com/DataDog/dd-sdk-ios/blob/60eae0ade10d9273f55636c38c6523085de6b339/DatadogCore/Sources/Core/DatadogCore.swift#L294-L296) to only `DatadogRemoteFeatures` (the ones that can write and read batch files). In consequence, it won't be available in following features:
- `CrashReportingFeature`
- `NetworkInstrumentationFeature`
- `BacktraceReportingFeature`

While last two are rather fine, there might be use cases for `DataStore` in Crash Reporting. I don't foresee any at this point, hence I don't consider it blocking.

Even if `DataStore` is required for `CrashReportingFeature` at some point, I rather thing we should work on the `FeatureScope` interface rather than re-design the `DataStore` management. I'm open for discussion 🙂.

Reply:
>  I rather thing we should work on the FeatureScope interface rather than re-design the DataStore management.

Totally 👍 

/suggestion can we use a `var` instead?